### PR TITLE
Create a fixupImports function as a docs replacement ruleset

### LINKED ISSUE #1592: Docs: automatically remove TF stock import caveat
### What happened?

@t0yv0 noticed that [`terraform` shows up in our GCP docs more then 600 times](https://github.com/pulumi/pulumi-gcp/issues/1473). Investigation shows that the block:

```
In Terraform v1.5.0 and later, use an [`import` block](https://developer.hashicorp.com/terraform/language/import) to import
```

alone shows up 502 times. This block should be stripped by default for all providers.

### Example

```json
        "gcp:accesscontextmanager/egressPolicy:EgressPolicy": {
            "description": "This resource has been deprecated, please refer to ServicePerimeterEgressPolicy.\n\n\nTo get more information about EgressPolicy, see:\n\n* [API documentation](https://cloud.google.com/access-context-manager/docs/reference/rest/v1/accessPolicies.servicePerimeters#egresspolicy)\n\n\n## Import\n\nEgressPolicy can be imported using any of these accepted formats* `{{egress_policy_name}}/{{resource}}` In Terraform v1.5.0 and later, use an [`import` block](https://developer.hashicorp.com/terraform/language/import) to import EgressPolicy using one of the formats above. For exampletf import {\n\n id = \"{{egress_policy_name}}/{{resource}}\"\n\n to = google_access_context_manager_egress_policy.default }\n\n```sh\n $ pulumi import gcp:accesscontextmanager/egressPolicy:EgressPolicy When using the [`terraform import` command](https://developer.hashicorp.com/terraform/cli/commands/import), EgressPolicy can be imported using one of the formats above. For example\n```\n\n\n\n```sh\n $ pulumi import gcp:accesscontextmanager/egressPolicy:EgressPolicy default {{egress_policy_name}}/{{resource}}\n```\n\n ",
            "properties": {
```

[(src)](https://github.com/pulumi/pulumi-gcp/blob/572df0de1f84f0b29f7fa1ff7d653ca1cf7734ee/provider/cmd/pulumi-resource-gcp/schema.json#L94485-L94488)

### Output of `pulumi about`

NA

### Additional context

AWS already handles this in a [systemic way](https://github.com/pulumi/pulumi-aws/blob/8fdc0c3c6c151e64203c0290683daea92ee945d8/provider/doc_edits.go#L70-L94). To get a bridge level fix, we should be able to hoist AWS's fix into the set of default edit rules:

https://github.com/pulumi/pulumi-terraform-bridge/blob/0f7ccfdef31d3fddd4eb46c9770d460bbfbe81e3/pkg/tfgen/docs.go#L190-L199

This will need tests to land, but those can also be lifted from AWS.

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

### PR DESCRIPTION
**Pull request by guineveresaenger**
This pull request creates a new DocsEdit rule that categorically removes import blocks of the format,
"In Terraform v1.5.0 and later, use an `import` block to import..." which cleans up unusable import
sections such as [this one](https://www.pulumi.com/registry/packages/gcp/api-docs/compute/instancegroupmanager/#import).

Of note: prior to this change, the link stripping was not covering the import link due to it having 
the string `/terraform/` in it, not `.terraform`. This change removes the `.` from the regular expression,
looking instead for all links containing `terraform`. As an additional bonus, for the GCP provider
this results in a significant improvement of docs field descriptions, as all "Beta" resource field descriptions
contained a link that was not properly stripped and hence were omitted from the schema for containing the
`terraform` string.

[You can see the schema for pulumi-gcp here and compare](https://github.com/pulumi/pulumi-gcp/blob/guin/docs-gen-updates/provider/cmd/pulumi-resource-gcp/schema.json).

Fixes #1592.

Remove dot from terraform link expression to be able to catch import links with the same expression.
Add new link formats to link test.

### CODE REVIEW

File: pkg/tfgen/docs_test.go [hunk 1]
```diff
@@ -1317,3 +1317,75 @@ func readlines(t *testing.T, file string) []string {
...
+		` + "```yaml" + `
+		foo: bar
+		` + "```\n",
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.text, func(t *testing.T) {
+			//t.Parallel()
```
Comment (iwahbe) [inline pkg/tfgen/docs_test.go:1384]: Did you forget to remove this?
```suggestion
			t.Parallel()
```

File: pkg/tfgen/docs.go [hunk 2]
```diff
@@ -157,13 +157,39 @@ func boundedReplace(from, to string) tfbridge.DocsEdit {
...
+		" use an `import` block.*?```.+?```\n")
+
+	return tfbridge.DocsEdit{
+		Path: "*",
+		Edit: func(_ string, content []byte) ([]byte, error) {
+			// Strip import blocks
+			content = blockImportRegexp.ReplaceAllLiteral(content, nil)
+			content = inlineImportRegexp.ReplaceAllFunc(content, func(match []byte) []byte {
+				match = bytes.ReplaceAll(match, []byte("terraform"), []byte("pulumi"))
+				match = bytes.ReplaceAll(match, []byte("Terraform"), []byte("pulumi"))
```
Comment (iwahbe) [inline pkg/tfgen/docs.go:176]: If TF is capitalized, the Pulumi should probably be capitalized too.
```suggestion
				match = bytes.ReplaceAll(match, []byte("Terraform"), []byte("Pulumi"))
```

File: pkg/tfgen/docs_test.go [hunk 3]
```diff
@@ -1317,3 +1317,75 @@ func readlines(t *testing.T, file string) []string {
...
+		id = "vatp-8012925589"
+		}` + "\n```\n" + `post text:
+		` + "```yaml" + `
+		foo: bar
+		` + "```\n",
+			`post text:
+		` + "```yaml" + `
+		foo: bar
+		` + "```\n",
+		},
```
Comment (iwahbe) [inline pkg/tfgen/docs_test.go:1362]: I had a really hard time finding the break between strings.
```suggestion
			text: "In Terraform v1.5.0 and later, use an `import` block to import Transfer Workflows using the `id`. For example:\n" +
				"\n" +
				"```terraform" + `
		import {
		to = aws_verifiedaccess_trust_provider.example
		id = "vatp-8012925589"
		}` + "\n```\n" + `post text:
		` + "```yaml" + `
		foo: bar
		` + "```\n",
			expected: `post text:
		` + "```yaml" + `
		foo: bar
		` + "```\n",
		},
```

**Review by iwahbe — APPROVED** [review]
LGTM.

Some stats:
```
𝛌 git diff master --stat -- provider/cmd/pulumi-resource-gcp/schema.json
provider/cmd/pulumi-resource-gcp/schema.json | 1493 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++-----------------------------------------------------------------------------------------
 1 file changed, 760 insertions(+), 733 deletions(-)
```

Implies that you added 27 docs and improved 733.

Terraform has gone from being mentioned 2866 to 637 times in the schema.

**Review by t0yv0 — APPROVED** [review]
🚢

**Pull request merged**